server.py

```diff
@@ -35,40 +35,46 @@
 ######################################################################
 @app.route('/')
 def index():
-    pets_url = request.base_url + "pets"
-    return jsonify(name='Pet Demo REST API Service',
-                   version='1.0',
-                   url=pets_url
+    recommendations_url = request.base_url + "recommendations"
+    return jsonify(name='Recommendations REST API Service',
+                   version='0.1',
+                   url=recommendations_url
                    ), HTTP_200_OK
 
 ######################################################################
 # LIST ALL PETS
 ######################################################################
-@app.route('/pets', methods=['GET'])
-def list_pets():
-    results = pets.values()
-    kind = request.args.get('kind')
-    if kind:
-        results = []
-        for key, value in pets.iteritems():
-            if value['kind'] == kind:
-                results.append(pets[key])
+@app.route('/recommendations', methods=['GET'])
+def list_recommendations():
+    """
+    Basically prints out all the recommendations.
+    """
 
-    return reply(results, HTTP_200_OK)
+    result_dict = {}
+    for prod_id in data:
+        recomm_list = []
+        for recomm in data[prod_id]['recommendations']:
+            recomm_list.append(recomm['name'])
+        result_dict[data[prod_id]['name']] = recomm_list
+    return reply(result_dict, HTTP_200_OK)
```

Comment:
I think you are right about the list should be simple. I do think product id is necessary, so that other people can make a GET request for specific item easily. 
In our case, for each product id, we are sending back a list of recommendations. This is somehow different from what I learned from other resources. If we consider the list of recommendation as a value for the key "prod id", I think we should return this value or omit it as a whole.